mkdocs/docs/how-to-release.md

```diff
@@ -243,3 +252,17 @@ Make sure to create a PR to update the [GitHub issues template](https://github.c
 ## Update the integration tests
 
 Ensure to update the `PYICEBERG_VERSION` in the [Dockerfile](https://github.com/apache/iceberg-python/blob/main/dev/Dockerfile).
+
+## Create a Github Release Note
+
+Input the tag as the newly approved release version (e.g. `0.7.0`) and set it to create the tag on publish. Pick the target commit version as the commit ID the release was approved on.
+
+![Generate Release Notes](assets/images/gen-release-notes.jpg)
+
+Then, select the previous release version as the **Previous tag** to use the diff between the two versions in generating the release notes.
+
+Create a new [Release Note](https://github.com/apache/iceberg-python/releases/new) on the iceberg-python Github repository by clicking on **Draft a New Release**.
```

Comment:
nit: do you think this can be moved to the first line? 

Maybe something like:
Create a [new Release Note](https://github.com/apache/iceberg-python/releases/new) on the iceberg-python Github repository

The link goes directly to the new release page